src/util/opaque.h

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include <cstdlib>
+#include <memory>
+#include <optional>
+#include <utility>
+#include <variant>
+
+namespace bpftrace::util {
+
+// OpaqueValue corresponds to a blob of type-erased memory. It owns its own
+// underlying memory, taking a copy if necessary. This allows it to be used
+// to read maps, rings, and other locations which may not be permanent.
+//
+// This class does not really provide safety beyond first-order memory safety.
+class OpaqueValue {
+private:
+  struct OwnedBuffer {
+    OwnedBuffer(char *data) : data(data) {};
+    ~OwnedBuffer()
```

Comment:
Do we need to delete the copy and move constructors for this to prevent accidental double frees?

Reply:
It's not strictly needed because this is a private internal class, but I'm always for explicitness and safety. Added.